sdk/src/wallet/utxo.ts

```diff
@@ -69,20 +70,28 @@ export async function createBitcoinPsbt(
     const addressInfo = getAddressInfo(fromAddress);
 
     // TODO: possibly, allow other strategies to be passed to this function
-    const utxoSelectionStrategy = 'default';
+    let utxoSelectionStrategy = 'default';
+    if (addressInfo.type === AddressType.p2tr) {
+        utxoSelectionStrategy = 'all';
+    }
 
     if (addressInfo.network === 'regtest') {
         throw new Error('Bitcoin regtest not supported');
     }
 
     // We need the public key to generate the redeem and witness script to spend the scripts
-    if (addressInfo.type === (AddressType.p2sh || AddressType.p2wsh)) {
+    if (
+        addressInfo.type === AddressType.p2sh ||
+        addressInfo.type === AddressType.p2wsh ||
+        addressInfo.type === AddressType.p2tr
+    ) {
         if (!publicKey) {
             throw new Error('Public key is required to spend from the selected address type');
         }
     }
 
     const esploraClient = new EsploraClient(addressInfo.network);
+    const ordinalsClient = new OrdinalsClient(addressInfo.network);
```

Comment:
we can not really check if taproot address has any inscriptions in outputs with `/address/:address` endpoint (it does not exist). [Check](https://ordinals-mainnet.gobob.xyz/address/bc1p9u077kplyqud8qyc404l2t5ud8jerwph6t77cd8xqvnan30lef5sqxfjhm)

Reply:
Hmm we really should use that if possible, can you check why that endpoint is not yet enabled? Maybe its enabled on an early version. My concern with the approach below is the many additional HTTP calls we need for each UTXO

Reply:
The latest code of ord [includes](https://github.com/ordinals/ord/blob/master/src/subcommand/server.rs#L166) /address endpoint. 
I do agree that making 1 call to an API is more efficient than `N` calls for each utxo. Will update

Reply:
Agree with Greg, we should use the ord endpoint. Were you able to resolve the issue with our endpoint?

Reply:
Changes are in my stash. At the moment /address endpoint is still not available. Nikolai said that he updated ord service with the latest changes from upstream.

Reply:
> Agree with Greg, we should use the ord endpoint. Were you able to resolve the issue with our endpoint?

Changes added:
- request to `/address/:address` endpoint
- update interfaces to conform v0.21.2 ordinals api